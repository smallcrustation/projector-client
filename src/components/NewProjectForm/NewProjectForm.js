import React from 'react'

import ProjectsApiService from '../../services/projects-api-service'

export default class NewProjectForm extends React.Component {
  state = {error: null}

  handleSubmit = async e => {
    e.preventDefault()
    this.setState({error: null})

    const { project_name, location, budget } = e.target
    const newProject = {
      project_name: project_name.value,
      location: location.value,
      budget_original: budget.value
    }

    try {
      const savedProject = await ProjectsApiService.postProjects(newProject)
      project_name.value = ''
      location.value = ''
      budget.value = ''
      console.log(savedProject)
      this.props.onSuccessfulProject()
    } catch (err) {
      this.setState({ error: err.error })
    }
  }

  render() {
    const {error} = this.state
    return (
      <div className="NewProjectForm">
        <h3>New Project</h3>
        <form onSubmit={this.handleSubmit}>
          <div role="alert">{error && <p className="error">{error}</p>}</div>

          <label htmlFor="project_name">Project Name</label>
          <input type="text" id="project_name" name="project_name" required/>

          <label htmlFor="location">Location</label>
          <input type="address" id="location" name="location" />

          <label htmlFor="budget">Budget</label>
          <input type="currency" id="budget" name="budget" required/>

          <input type="submit" value="Submit" />
        </form>
      </div>
    )
  }
}
