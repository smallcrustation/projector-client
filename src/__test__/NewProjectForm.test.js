import React from 'react'
import ReactDOM from 'react-dom'

import NewProjectForm from '../components/NewProjectForm/NewProjectForm'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<NewProjectForm />, div)

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer.create(<NewProjectForm />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
