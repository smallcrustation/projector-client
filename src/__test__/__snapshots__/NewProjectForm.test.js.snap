// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `
<div
  className="NewProjectForm"
>
  <h3
    className="low-header"
  >
    New Project
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <label
      htmlFor="project_name"
    >
      Project Name
    </label>
    <input
      id="project_name"
      name="project_name"
      required={true}
      type="text"
    />
    <label
      htmlFor="location"
    >
      Location
    </label>
    <input
      id="location"
      name="location"
      required={true}
      type="address"
    />
    <label
      htmlFor="budget"
    >
      Budget
    </label>
    <input
      className="dollar-input"
      id="budget"
      name="budget"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      placeholder="$0.00"
      required={true}
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
